src/main/java/com/snowplowanalytics/snowplow/tracker/payload/TrackerEvent.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2020-2020 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.tracker.payload;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import com.snowplowanalytics.snowplow.tracker.Subject;
+import com.snowplowanalytics.snowplow.tracker.Tracker;
+import com.snowplowanalytics.snowplow.tracker.constants.Constants;
+import com.snowplowanalytics.snowplow.tracker.constants.Parameter;
+import com.snowplowanalytics.snowplow.tracker.events.*;
+
+/**
+ * A TrackerEvent which allows the TrackerPayload to be filled later. The payload will be
+ * filled by the Emitter in the Emitter thread, using the getTrackerPayload() method.
+ */
+public class TrackerEvent {
+
+    private final Event event;
+    private final Tracker tracker;
```

Comment:
Instead of giving it a `Tracker` as a property, did you consider giving it a `Subject` and a parameter `base64Encoded` (boolean)?  I think you use the tracker only to get those properties.  It feels like this class has been given access to method calls that it shouldn't really have access to.

Reply:
Agreed. There are a few other parameters too, where it calls back to Tracker to add them to the Payload but you're right, this doesn't feel right. I'll pass the required parameters in to this class on contruction. This will also remove the chance of mutating these parameters in the Tracker before the payload is built.